scripts/lib/aws/elasticache.sh

```diff
@@ -239,6 +307,59 @@ aws_wait_replication_group_deleted() {
   k8s_controller_reload_credentials "$service_name"
 }
 
+# aws_get_replication_group_json returns the JSON description of the replication group of interest
+# aws_get_replication_group_json requires 1 arguments:
+#    replication_group_id
+aws_get_replication_group_json() {
+  if [[ $# -ne 1 ]]; then
+    echo "FATAL: Wrong number of arguments passed to ${FUNCNAME[0]}"
+    echo "${FUNCNAME[0]} replication_group_id"
+    exit 1
+  fi
+  echo $(daws elasticache describe-replication-groups --replication-group-id "$1" | jq -r -e ".ReplicationGroups[0]")
+}
+
+# aws_get_rg_param_group asserts that the name of the parameter group associated with the provided replication group
+#   matches the name of the expected parameter group
+# aws_get_rg_param_group requires 2 arguments:
+#   replication_group_id
+#   expected_parameter_group_name
+aws_assert_rg_param_group() {
+  if [[ $# -ne 2 ]]; then
+    echo "FATAL: Wrong number of arguments passed to ${FUNCNAME[0]}"
+    echo "${FUNCNAME[0]} replication_group_id expected_parameter_group_name"
+    exit 1
+  fi
+
+  local primary_cluster=$(aws_get_replication_group_json "$1" | jq -r -e ".MemberClusters[0]")
+  local cluster_json=$(daws elasticache describe-cache-clusters --cache-cluster-id "$primary_cluster" | jq -r -e ".CacheClusters[0]")
+  local param_group=$(echo $cluster_json | jq -r -e ".CacheParameterGroup .CacheParameterGroupName")
+  if [[ "$param_group" != "$2" ]]; then
+    echo "FAIL: expected replication group $1 to have parameter group $2. Actual: $param_group"
+    exit 1
+  fi
+}
+
+# aws_assert_replication_group_property compares the requested property, retrieved from the AWS CLI,
+#   to the expected value of that property.
+# aws_assert_replication_group_property requires 3 arguments:
+#   replication_group_id
+#   jq_filter – the property of interest, e.g. ".CacheNodeType"
+#   expected_value
+aws_assert_replication_group_property() {
+  if [[ $# -ne 3 ]]; then
+    echo "FATAL: Wrong number of arguments passed to ${FUNCNAME[0]}"
+    echo "${FUNCNAME[0]} replication_group_id jq_filter expected_value"
+    exit 1
+  fi
+  local actual_value=$(aws_get_replication_group_json "$1" | jq -r -e "$2")
+  if [[ "$3" != "$actual_value" ]]; then
+    echo "FAIL: property $2 for replication group $1 has value '$actual_value', but expected '$3'"
+    print_k8s_ack_controller_pod_logs
+    exit 1
+  fi
```

Comment:
I actually prefer the style of having named local variables instead of "$1", "$2", "$3", etc... makes the function much easier to read, IMHO.

Reply:
Yeah agreed, I was trying to shorten functions but the loss in readability is not really worth it